specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json

```
@@ -551,31 +550,395 @@
+              "Microsoft.KeyVault/managedHsm/backup/start/action",
+              "Microsoft.KeyVault/managedHsm/restore/start/action",
+              "Microsoft.KeyVault/managedHsm/backup/status/action",
+              "Microsoft.KeyVault/managedHsm/restore/status/action"
+            ],
+            "x-ms-enum": {
+              "name": "AllowedDataActions",
+              "modelAsString": true,
+              "values": [
+                {
+                  "name": "ReadHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/read/action",
+                  "description": "Read HSM key."
+                },
+                {
+                  "name": "WriteHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/write/action",
+                  "description": "Create or update HSM key."
+                },
+                {
+                  "name": "ReadDeletedHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/deletedKeys/read/action",
+                  "description": "Read deleted HSM key."
+                },
+                {
+                  "name": "RecoverDeletedHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/deletedKeys/recover/action",
+                  "description": "Recover deleted HSM key."
+                },
+                {
+                  "name": "BackupHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/backup/action",
+                  "description": "Backup HSM keys."
+                },
+                {
+                  "name": "RestoreHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/restore/action",
+                  "description": "Restore HSM keys."
+                },
+                {
+                  "name": "DeleteRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/delete/action",
+                  "description": "Delete role assignment."
+                },
+                {
+                  "name": "GetRoleRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/read/action",
+                  "description": "Get role assignment."
+                },
+                {
+                  "name": "WriteRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/write/action",
+                  "description": "Create or update role assignment."
+                },
+                {
+                  "name": "ReadRoleDefinition",
+                  "value": "Microsoft.KeyVault/managedHsm/roleDefinitions/read/action",
+                  "description": "Get role definition."
+                },
+                {
+                  "name": "EncryptHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/encrypt/action",
+                  "description": "Encrypt HSM keys."
```

```suggestion
                  "description": "Encrypt using HSM key."
```

That should probably change the name as well (same below), but no great idea here. Maybe "HsmKeyEncrypt"? In fact, since these are already namespaced to Managed HSM and keys, and, hypothetically, we could add more, distinct ones for KV in the future, should we maybe use names like "HsmKey<Permission>" throughout?

---

Every key action should be named like "KeyEncrypt", "KeyDecrypt" for consistency with AKV / MHSM docs - in addition all of the key actions with the exception of import are using HSM keys, like Heath said

---

I renamed a bunch of them - please take another look and let me know if I they now all align with what you were thinking.